lib/MetaCPAN/Model/Tarball.pm

```diff
@@ -0,0 +1,177 @@
+package MetaCPAN::Model::Tarball;
+
+use Archive::Any;
+use CPAN::DistnameInfo ();
+use DateTime           ();
+use DDP;
+use Log::Contextual qw( :log :dlog );
+use MetaCPAN::Types qw(ArrayRef File Dir);
+use Moose;
+use MooseX::StrictConstructor;
+use Path::Class qw(file dir);
+
+has tarball => (
+    is       => 'ro',
+    isa      => File,
+    required => 1,
+    coerce   => 1,
+);
+
+has files => (
+    is       => 'ro',
+    isa      => ArrayRef,
+    init_arg => undef,
+    lazy     => 1,
+    builder  => '_build_files'
+);
+
+has base_dir => (
+    is      => 'ro',
+    isa     => Dir,
+    coerce  => 1,
+    default => '/tmp',
+);
+
+has date => (
+    is  => 'ro',
+    isa => 'DateTime',
+);
+
+has index => (
+    is         => 'ro',
+    lazy_build => 1,
+);
+
+has author => (
+    is         => 'ro',
+    lazy_build => 1,
+);
+
+has name => (
+    is         => 'ro',
+    lazy_build => 1,
+);
+
+has metadata => (
+    is         => 'ro',
+    lazy_build => 1,
+);
+
+has status => (
+    is         => 'ro',
+    lazy_build => 1,
+);
+
+has distribution => (
+    is         => 'ro',
+    lazy_build => 1,
+);
+
+has version => (
+    is         => 'ro',
+    lazy_build => 1,
+);
+
+has maturity => (
+    is         => 'ro',
+    lazy_build => 1,
+);
+
+has release => (
+    is         => 'ro',
+    lazy_build => 1,
+);
+
+sub _build_files {
+    my $self = shift;
+    log_info {"Processing $self->tarball"};
+
+    my $at = Archive::Any->new( $self->tarball );
+    my $tmpdir
+        = dir( File::Temp::tempdir( CLEANUP => 0, DIR => $self->base_dir ) );
+    log_error {"$self is being impolite"} if $at->is_impolite;
+
+    log_error {"$self is being naughty"} if $at->is_naughty;
+
+    log_debug {"Extracting archive to filesystem"};
+    $at->extract($tmpdir);
+
+    my @files;
+    my @list = $at->files;
```

Comment:
`@list` is pretty non-descriptive too, but you only use it once, so you don't need to create the variable.

```
log_debug { 'Indexing ', scalar $at->files, " files" };
```